concepts/modularization/migration.md

```diff
@@ -0,0 +1,32 @@
+# Migration plan
+
+Migration plan will be executed several times until all modules are moved from reconciler to lifecycle-manager. Is it possible to migrate several modules at once but it is not recommended. The migration plan uses `serverless` module as an example.
+
+The prerequisite for migration is that lifecycle-manager is already activated in KCP for all the clusters (Kyma CRs are created in the KCP and SKR). Migration is split into 2 phases: enabling module for the new clusters and update existing clusters.
+
+![migration](assets/migration.drawio.svg)
+
+## Enable module for new clusters (first)
+
+First step is to enable module installation with lifecycle manager for new clusters. Existing clusters are still managed by reconciler.
+- Disable module template replication for existing clusters (moduleCatalog=false). This prevents users to add serverless module before it is disabled in the reconciler.
+- New Kyma version is enabled in KEB (without serverless module). New clusters do not have serverless until customer enabled it in Kyma resource.
+- Serverless operator(manager) is ready and module template is available in the fast/regular channel. 
+
+## Update existing clusters
+
+Migration can be stopped at any moment if any problem occurs. Not migrated clusters will still use serverless reconciler (old Kyma version with serverless)
+
+- Upgrade the clusters to the new Kyma version (without serverless) - from this point reconciler doesn't touch serverless resources.
+- Add module to Kyma CR in SKR with option to not copy default CR. Lifecycle manager will install serverless manager without starting module installation. Default CR creation is omitted for the case when we need different setting for different clusters (eg. trial / production). If default CR can be applied for all the clusters then module copy can stay enabled and the next step is not required.
+- Create CRD  and default CR for serverless manager in the SKR. CRD creation is optional - it can be skipped, but then migration script has to wait until liefecycle-manager installs it. Entire step can be skipped if the default module CR can be applied for all the clusters.
```

Comment:
At this point, the Kyma CR is expected to be synced back to the KCP, right, so it will reflect that serverless has been enabled on the SKR?

Reply:
To be precise Kyma CR spec from the runtime cluster is not copied back to the control plane, but you can see that serverless is enabled in the status.